Resources/Prototypes/Entities/Mobs/NPCs/animals.yml

```diff
@@ -2749,3 +2749,61 @@
     factions:
     - Passive
 
+- type: entity
+  name: armadillo
+  parent: SimpleMobBase
+  id: MobArmadillo
+  description: some sort of armored rat.
+  components:
+  - type: MovementSpeedModifier
+    baseWalkSpeed: 4
+    baseSprintSpeed: 6
+  - type: Sprite
+    drawdepth: Mobs
+    layers:
+    - map: ["enum.DamageStateVisualLayers.Base"]
+      state: armadillo
+      sprite: Mobs/Animals/armadillo.rsi
+  - type: Physics
+  - type: Fixtures
+    fixtures:
+      fix1:
+        shape:
+          !type:PhysShapeCircle
+          radius: 0.25
+        density: 40
+        mask:
+        - MobMask
+        layer:
+        - MobLayer
+  - type: DamageStateVisuals
+    states:
+      Alive:
+        Base: armadillo
+      Critical:
+        Base: ball
```

Comment:
the crit state being a ball is kinda strange. If we have support for walk animations, then the ball rolling would be good. I'm not too sure where to look for that, however.

Reply:
I was unable to find it if it exists, that was my original idea.